bigchaindb/core.py

```diff
@@ -274,8 +277,27 @@ def get_status(self, txid):
             or 'backlog'). If no transaction with that `txid` was found it
             returns `None`
         """
-        _, status = self.get_transaction(txid, include_status=True)
-        return status
+        try:
+            statuses = self.get_blocks_status_containing_tx(txid).values()
+        except (TypeError, AttributeError):
+            logger.debug(
+                'No blocks found for transaction %s. Looking into BACKLOG.',
+                txid,
+            )
+            if backend.query.\
+                    get_transaction_from_backlog(self.connection, txid):
+                return self.TX_IN_BACKLOG
```

Comment:
There are two cases for which we check the **`backlog`** table:

1. The transaction is not in the blocks table (**`bigchain`**). That is, it is not in any block (`UNDECIDED`, `VALID`, `INVALID`), so we check the **`backlog`** table as a last attempt to find it.
2. The transaction is in the blocks table (**`bigchain`**), but only in an `INVALID` block. Since it may have been sent back to the **`backlog`** to be processed again, so we check the **`backlog`**. If it is not found in the **`backlog`** we will then fall back on the `INVALID` state.